metacat-connector-druid/src/main/java/com/netflix/metacat/connector/druid/DruidConnectorTableService.java

```diff
@@ -66,9 +71,21 @@ public DruidConnectorTableService(
     @Override
     public TableInfo get(@Nonnull final ConnectorRequestContext context, @Nonnull final QualifiedName name) {
         log.debug("Get table metadata for qualified name {} for request {}", name, context);
-        final ObjectNode node = this.druidClient.getLatestDataByName(name.getTableName());
-        final DataSource dataSource = DruidConverterUtil.getDatasourceFromLatestSegmentJsonObject(node);
-        return this.druidConnectorInfoConverter.getTableInfoFromDatasource(dataSource);
+        try {
+            final ObjectNode node = this.druidClient.getLatestDataByName(name.getTableName());
+            final DataSource dataSource = DruidConverterUtil.getDatasourceFromLatestSegmentJsonObject(node);
+            return this.druidConnectorInfoConverter.getTableInfoFromDatasource(dataSource);
+        } catch (MetacatException e) {
```

Comment:
Any metacat exception is rethrown as a TableNotFound?

Reply:
Method getLatestDataByName throws this specific error when segments are not found.